websites/A/Armorgames/presence.ts

```diff
@@ -0,0 +1,113 @@
+const presence = new Presence({
+  clientId: "827910536049852488"
+}), browsingStamp = Math.floor(Date.now() / 1000);
+let search: HTMLInputElement,
+title: HTMLElement;
+
+presence.on("UpdateData", async () => {
+  const presenceData: PresenceData = {
+    largeImageKey: "logo"
+  },
+  page = window.location.pathname;
+
+    presenceData.startTimestamp = browsingStamp;
+    if (page === "/") {
+      search = document.querySelector("#search-input");
+      if (!search) {
+        presenceData.details = "Viewing The Homepage";
+      } else if (search.value !== "") {
+        presenceData.details = "Searching:";
+        presenceData.state = search.value;
+        presenceData.smallImageKey = "searching";
+      } else { 
+        presenceData.details = "Viewing the homepage";
+      }
+    } else if (page.includes("/category/")) {
+      if (page == "/category/all") {
+        presenceData.details = "Viewing All Categories";
+      } else {
+        title = document.querySelector("#content-canvas > div.cap > div > h2 > a > span");
+        presenceData.details = "Viewing Category:";
+        presenceData.state = title.textContent;
+        presenceData.buttons = [{label: "View Category", url: window.location.href}];
+      }
+    } else if (page == "/community") {
+      presenceData.details = `Viewing the forums`;
+    } else if (page.includes("/forum/")) {
+      title = document.querySelector("#content-canvas > main > h1");
+      presenceData.details = `Viewing Forum:`;
+      presenceData.state = title.textContent.replace("Forums → ", "");
+      presenceData.buttons = [{label: "View Forum", url: window.location.href}];
+    } else if (page.includes("thread")) {
+      title = document.querySelector("#thread-title");
+      presenceData.details = `Reading Forum Thread:`;
+      presenceData.state = title.textContent;
+      presenceData.buttons = [{label: "View Thread", url: window.location.href}];
+    } else if (page.includes("/friends/")) {
+      title = document.querySelector("#main > h2");
+      if (title.textContent == "Your Friends") {
+        presenceData.details = `Viewing my friends`;
+        presenceData.buttons = [{label: "View Friends", url: window.location.href}];
```

Comment:
Thier?

Reply:
????

Reply:
I mean "my" is the one who using Presence right, so it should be "Their friends"